Source/Scene/PointPrimitive.js

```diff
@@ -0,0 +1,488 @@
+            get : function() {
+                return this._pickPrimitive;
+            },
+            set : function(value) {
+                this._pickPrimitive = value;
+                if (defined(this._pickId)) {
+                    this._pickId.object.primitive = value;
+                }
+            }
+        }
+    });
+
+    PointPrimitive.prototype.getPickId = function(context) {
+        if (!defined(this._pickId)) {
+            this._pickId = context.createPickId({
+                primitive : this._pickPrimitive,
+                collection : this._collection,
+                id : this._id
+            });
+        }
+
+        return this._pickId;
+    };
+
+    PointPrimitive.prototype._getActualPosition = function() {
+        return this._actualPosition;
+    };
+
+    PointPrimitive.prototype._setActualPosition = function(value) {
+        Cartesian3.clone(value, this._actualPosition);
+        makeDirty(this, POSITION_INDEX);
+    };
+
+    var tempCartesian3 = new Cartesian4();
+    PointPrimitive._computeActualPosition = function(position, frameState, modelMatrix) {
+        if (frameState.mode === SceneMode.SCENE3D) {
+            return position;
+        }
+
+        Matrix4.multiplyByPoint(modelMatrix, position, tempCartesian3);
+        return SceneTransforms.computeActualWgs84Position(frameState, tempCartesian3);
+    };
+
+    var scratchMatrix4 = new Matrix4();
+    var scratchCartesian4 = new Cartesian4();
+
+    PointPrimitive._computeScreenSpacePosition = function(modelMatrix, position, scene) {
+        // This function is basically a stripped-down JavaScript version of PointPrimitiveCollectionVS.glsl
+        var camera = scene.camera;
+        var view = camera.viewMatrix;
+        var projection = camera.frustum.projectionMatrix;
+
+        // Model to eye coordinates
+        var mv = Matrix4.multiplyTransformation(view, modelMatrix, scratchMatrix4);
+        var positionEC = Matrix4.multiplyByVector(mv, Cartesian4.fromElements(position.x, position.y, position.z, 1, scratchCartesian4), scratchCartesian4);
+        var positionCC = Matrix4.multiplyByVector(projection, positionEC, scratchCartesian4); // clip coordinates
+        var positionWC = SceneTransforms.clipToGLWindowCoordinates(scene, positionCC, new Cartesian2());
+
+        return positionWC;
+    };
+
+    /**
+     * Computes the screen-space position of the pointPrimitive's origin.
+     * The screen space origin is the top, left corner of the canvas; <code>x</code> increases from
+     * left to right, and <code>y</code> increases from top to bottom.
+     *
+     * @param {Scene} scene The scene.
+     * @returns {Cartesian2} The screen-space position of the pointPrimitive.
+     *
+     * @exception {DeveloperError} PointPrimitive must be in a collection.
+     *
+     * @example
+     * console.log(p.computeScreenSpacePosition(scene).toString());
+     */
+    PointPrimitive.prototype.computeScreenSpacePosition = function(scene) {
```

Comment:
Use a result parameter to avoid the `new Cartesian2` above?  Bonus if you add a warning to the other `computeScreenSpacePosition` and deprecate the non-result usage.